### ISSUE TITLE
Add a parameter to control ordering on list requests

### ISSUE THREAD

**Issue by Mr0grog** [edgi-govdata-archiving/web-monitoring-db#165]
We should have a relatively standard way to control ordering on requests that return a list of pages/versions/changes/annotations. This could be as simple as ascending/descending or as complex as a list of fields and sort directions, e.g. `capture_time:desc,title:asc` or something.

This needs a clearer design proposal before implementation.

**Comment by tylerferraro** [comment]
I have seen two common RESTful sorting standards.

1) `?sort=title:desc,capture_time:asc`
2) `?sort=-title,capture_time`

The first is more readable and easy to understand. The second uses a `-` to denote descending order and everything else defaults to ascending.

I wrote up a potential solution into a PR on my forked repository for you to review so we can discuss how best to solve this issue. https://github.com/tylerferraro/web-monitoring-db/pull/1

**Comment by danielballan** [comment]
I'll throw in a drive-by :+1: for (1).

**Comment by Mr0grog** [comment]
Adding this to v0.0.2—I realized after doing some big queries that involved iterating through many requests recently that we have two really critical sort-related problems:

- We have some controllers with no explicit sorting at all, which is real problem because that means they are sorting by primary key, which is a UUID, which is not sortable, so when iterating through many result chunks, you have no guarantee that you are getting a consistent part of the results on any given request. We need a default sort on all list views.

- Most list views that *do* have explicit sorting are sorted by `updated_at` descending, which means that if your query takes long enough (e.g. you want to list all pages, which is needed for edgi-govdata-archiving/web-monitoring-processing#86 — take note, @weatherpattern), you are likely to have the results shuffle around on you because pages were updated *during your query.* So the only way to guarantee a consistent full set of results in an extremely large query is to sort by `created_at` ascending. Therefore, we need customized sorting.
